### ISSUE TITLE
Add support for HTTP1 communications to the Networking API

### ISSUE THREAD

**Issue by Duckwhale** [evo-lua/evo-luvi#137]
Goals:

- [ ] Can perform HTTP requests (``HttpClient`` module)
- [ ] Can create HTTP servers (``HttpServer`` module)
- [ ] Spec compliance is tested for the latest HTTP 1 RFCs
  - [ ] [RFC 9110: HTTP Semantics](https://www.rfc-editor.org/rfc/rfc9110)
  - [ ] [RFC 9112: HTTP/1.1](https://www.rfc-editor.org/rfc/rfc9112)
- [ ] Performance should be at least acceptable (benchmarks indicate 1 GB/s for native llhttp/ffi and 20-25 MB/s for luvit)
  - [ ] Great: 80% or higher (almost certainly impossible unless all buffering/Lua structures are disabled)
  - [ ] Good: 60% or higher (must use cdata only with a native FFI layer, and no Lua object creation)
  - [ ] Mediocre: 20% or higher (buffering and Lua object creation with native FFI layer)
  - [ ] Mandatory: Must be (much) better than the 2.5% or so that a pure Lua implementation can do (e.g, luvit ``http`` module)

Non-goals:

- [ ] TLS/HTTPS support (separate issue)
- [ ] Streaming APIs (required for large files): Buffering is easier to use and sufficient for now, the rest can be built on top
- [ ] Caching (RFC 9111): Not really necessary, nginx/apache are likely better solutions where this is needed

**Comment by Duckwhale** [comment]
This is just to formalize the requirements after building a couple of different prototypes. Requires #90 (mostly done, needs cleanup).

As for performance, I am using llhttp-ffi as the maximum (1 GB/s on my desktop) and luvit (pure Lua, measured at 20-25 MB/s on the same machine) as the minimum. It is clear that the requirement of providing a user-friendly API is directly at odds with reducing overhead, and a streaming API on top of that will also be less user-friendly by default.

For this issue, I will limit the scope to a basic, buffering and somewhat inefficient implementation that is sufficient to proceed with the WebSocket protocol support. Streaming can be performed by simply overriding the event handlers (and then creating async write requests via libuv), while reducing the memory footprint is possible by disabling the FFI layer's event buffering (which comes with other drawbacks). I doubt there's a perfect solution, or even one that fulfills most of the requirements, here.

**Comment by rdw-software** [comment]
Blocked by #168, since hacking in the URL parsing logic required for some of the RFC specs doesn't seem like a good idea.
